base/Base.jl

```diff
@@ -108,12 +108,24 @@ include("options.jl")
 function Core.kwcall(kwargs, ::typeof(invoke), f, T, args...)
     @inline
     # prepend kwargs and f to the invoked from the user
-    T = rewrap_unionall(Tuple{Any, Core.Typeof(f), (unwrap_unionall(T)::DataType).parameters...}, T)
+    T = rewrap_unionall(Tuple{Core.Typeof(kwargs), Core.Typeof(f), (unwrap_unionall(T)::DataType).parameters...}, T)
     return invoke(Core.kwcall, T, kwargs, f, args...)
 end
 # invoke does not have its own call cache, but kwcall for invoke does
 setfield!(typeof(invoke).name.mt, :max_args, 3, :monotonic) # invoke, f, T, args...
 
+# define applicable(f, T, args...; kwargs...), without kwargs wrapping
+# to forward to applicable
+function Core.kwcall(kwargs, ::typeof(applicable), @nospecialize(args...))
+    @inline
+    return applicable(Core.kwcall, kwargs, args...)
+end
```

Comment:
This definition doesn't account for keyword arguments that aren't declared?
```julia
julia> fkw(a::Int; kw=nothing) = isnothing(kw) ? 0 : a
fkw (generic function with 1 method)

julia> applicable(fkw, 10; kw=20)
true

julia> applicable(fkw, 10; kw2=20)
true # shouldn't this be `false`?
```

Reply:
It should be true, since kwarg presence/absence does not participate in or affect dispatch.

Though it you want to be horrible, I suppose you could declare the function type `@NamedTuple(a::Int)` to require specific kwarg orders in dispatch. Though it would be unsupported and unstable.

Reply:
Yes, but I wondered if we want to make it consistent with the behavior of `hasmethods`.